analyzers/tests/SonarAnalyzer.UnitTest/Rules/CallerInformationParametersShouldBeLastTest.cs

```diff
@@ -25,26 +25,33 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class CallerInformationParametersShouldBeLastTest
     {
+        private readonly VerifierBuilder verifier = new VerifierBuilder<CallerInformationParametersShouldBeLast>();
+
         [TestMethod]
         public void CallerInformationParametersShouldBeLast() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\CallerInformationParametersShouldBeLast.cs", new CallerInformationParametersShouldBeLast());
+            verifier.AddPaths("CallerInformationParametersShouldBeLast.cs").Verify();
 
 #if NET
+
         [TestMethod]
         public void CallerInformationParametersShouldBeLast_CSharp9() =>
-            OldVerifier.VerifyAnalyzerFromCSharp9Console(@"TestCases\CallerInformationParametersShouldBeLast.CSharp9.cs", new CallerInformationParametersShouldBeLast());
+            verifier.AddPaths("CallerInformationParametersShouldBeLast.CSharp9.cs").WithTopLevelStatements().Verify();
+
+        [TestMethod]
+        public void CallerInformationParametersShouldBeLast_CSharp10() =>
+            verifier.AddPaths("CallerInformationParametersShouldBeLast.CSharp10.cs").WithOptions(ParseOptionsHelper.FromCSharp10).Verify();
 
         [TestMethod]
         public void CallerInformationParametersShouldBeLast_CSharpPreview() =>
-            OldVerifier.VerifyAnalyzerCSharpPreviewLibrary(@"TestCases\CallerInformationParametersShouldBeLast.CSharpPreview.cs", new CallerInformationParametersShouldBeLast());
+            verifier.AddPaths("CallerInformationParametersShouldBeLast.CSharpPreview.cs").WithOptions(ParseOptionsHelper.CSharpPreview).Verify();
+
 #endif
 
         [TestMethod]
         public void CallerInformationParametersShouldBeLastInvalidSyntax() =>
-            new VerifierBuilder<CallerInformationParametersShouldBeLast>()
-                .AddPaths("CallerInformationParametersShouldBeLastInvalidSyntax.cs")
-                .WithErrorBehavior(CompilationErrorBehavior.Ignore)
-                .WithConcurrentAnalysis(false)
-                .VerifyNoIssueReported();
+            verifier.AddPaths("CallerInformationParametersShouldBeLastInvalidSyntax.cs")
+                    .WithErrorBehavior(CompilationErrorBehavior.Ignore)
+                    .WithConcurrentAnalysis(false)
+                    .Verify();
```

Comment:
`VerifyNoIssueReported` was changed to `Verify`. There was one test case marked as `Noncompliant`:

`public void Method6([CallerLineNumber][CallerFilePath]string callerFilePath, string other) { } // Noncompliant`

Because of `VerifyNoIssueReported` the `// Noncompliant` was ignored. Now the `// Noncompliant` is verified.

The root cause is, that most of the method declarations are parsed as local functions.